src/client/chat.js

```diff
@@ -16,6 +17,25 @@ function isFormatted (message) {
   }
 }
 
+// Used for 1.20.3+ to convert NBT back into the JSON string expected of a chat message.
+function handleNBTStrings (nbtDataOrString) {
```

Comment:
should this function be exposed? I think there are now dozen of packets using this nbt thing